docs/api/api_docs.yaml

```diff
@@ -204,61 +731,244 @@ paths:
+                properties:
+                  _id:
+                    type: string
+                    description: The ID of the document.
+                  _rev:
+                    type: string
+                    description: The revision ID of the document.
+              examples:
+                Example with no optional fields filled:
+                  value:
+                    FailedLoginAttempts: 5
+                    Friends:
+                      - Bob
+                    _id: AliceSettings
+                    _rev: 1-64d4a1f179db5c1848fe52967b47c166
+        '400':
+          $ref: '#/components/responses/invalid-doc-id'
+        '404':
+          $ref: '#/components/responses/Not-found'
+        '501':
+          description: Not Implemented. It is likely this error was caused due to trying to use an enterprise-only feature on the community edition.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTP-Error'
       tags:
         - Admin
         - Public
-    head:
+      summary: Get a document
+      parameters:
+        - $ref: '#/components/parameters/rev'
+        - $ref: '#/components/parameters/open_revs'
+        - $ref: '#/components/parameters/show_exp'
+        - $ref: '#/components/parameters/revs_from'
+        - $ref: '#/components/parameters/atts_since'
+        - $ref: '#/components/parameters/revs_limit'
+        - $ref: '#/components/parameters/includeAttachments'
+        - $ref: '#/components/parameters/replicator2'
+      description: Retrieve a document from the database by it's doc ID.
+    put:
       responses:
-        '200':
-          description: OK
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/New-revision'
+          headers:
+            Etag:
+              schema:
+                type: string
+              description: The revision of the upserted document. Not set if query option `new_edits` is true.
+        '400':
+          $ref: '#/components/responses/request-problem'
+        '404':
+          $ref: '#/components/responses/Not-found'
+        '409':
+          $ref: '#/components/responses/Conflict'
+        '415':
+          $ref: '#/components/responses/Invalid-content-type'
       tags:
         - Admin
         - Public
-    put:
+      description: |-
+        This will upsert a document meaning if it does not exist, then it will be created. Otherwise a new revision will be made for the existing document. A revision ID must be provided if targetting an existing document.
+
+        A document ID must be specified for this endpoint. To let Sync Gateway generate the ID, use the `POST /{db}/` endpoint. 
+
+        If a document does exist, then only the fields specified in the body will be changed. This means unspecified fields will not be mutated but will be in the new revision.
```

Comment:
As far as I know we _never_ do this merge/upsert thing for document bodies. A PUT is a full-replacement of the doc body.

Reply:
Yes, you are right. Thought that upsert meant that if it does not exist, it is created. Since the case is that it means a merge is done if it does exist, I will change `PUT /{db}/_user/{name}` to reflect this as well.